src/views/Ifos/hooks/v1/useGetPublicIfoData.ts

```diff
@@ -74,9 +73,43 @@ const useGetPublicIfoData = (ifo: Ifo): PublicIfoData => {
     }))
   }, [address, currentBlock, releaseBlockNumber])
 
+  const fetchInActiveIfoData = useCallback(async () => {
+    const ifoCalls = ['startBlock', 'endBlock', 'raisingAmount', 'totalAmount'].map((method) => ({
+      address,
+      name: method,
+    }))
+
+    const [startBlock, endBlock, raisingAmount, totalAmount] = await multicallv2(ifoV1Abi, ifoCalls)
+
+    const startBlockNum = startBlock ? startBlock[0].toNumber() : 0
+    const endBlockNum = endBlock ? endBlock[0].toNumber() : 0
+
+    const status = 'finished' as IfoStatus
+
+    setState((prev) => ({
+      ...prev,
+      status,
+      startBlockNum,
+      endBlockNum,
+      [PoolIds.poolUnlimited]: {
+        ...prev.poolUnlimited,
+        raisingAmountPool: raisingAmount ? new BigNumber(raisingAmount[0].toString()) : BIG_ZERO,
+        totalAmountPool: totalAmount ? new BigNumber(totalAmount[0].toString()) : BIG_ZERO,
+      },
+    }))
+  }, [address])
+
+  useEffect(() => {
+    if (ifo.isActive) {
+      fetchIfoData()
+    }
+  }, [fetchIfoData, ifo])
+
   useEffect(() => {
-    fetchIfoData()
-  }, [fetchIfoData])
+    if (!ifo.isActive) {
+      fetchInActiveIfoData()
+    }
+  }, [fetchInActiveIfoData, ifo])
```

Comment:
Perhaps this can be 1 hook?
```js
if (ifo.isActive) {
  fetchIfoData()
} else {
  fetchInActiveIfoData()
}
```

Reply:
But that means everytime fetchIfoData changes we will call fetchInActiveIfoData